src/util/ConfigManager/ConfigManager.h

```diff
@@ -0,0 +1,247 @@
+  key must be string, equal to the name of the configuration option.
+  */
+  void addConfigOption(const VectorOfKeysForJson& pathToOption,
+                       ConfigOption&& option);
+
+  // For testing.
+  FRIEND_TEST(ConfigManagerTest, GetConfigurationOptionByNestedKeysTest);
+  FRIEND_TEST(ConfigManagerTest, ParseConfig);
+  FRIEND_TEST(ConfigManagerTest, ParseConfigExceptionTest);
+  FRIEND_TEST(ConfigManagerTest, ParseShortHandTest);
+
+ public:
+  /*
+  @brief Creates and adds a new configuration option.
+
+  @tparam OptionType The type of value, the configuration option can hold.
+
+  @param pathToOption Describes a path in json, that points to the value held by
+  the configuration option. The last key in the vector is the name of the
+  configuration option. NOTE: Both the first, and the last key have to be
+  strings.
+  @param optionDescription A description for the configuration option.
+  @param variableToPutValueOfTheOptionIn The value held by the configuration
+  option will be copied into this variable, whenever the value in the
+  configuration option changes.
+  @param defaultValue A default value for the configuration option. If none is
+  given, signified by an empty optional, then a value for the configuration
+  option MUST be given at runtime.
+  */
+  template <typename OptionType>
+  requires ad_utility::isTypeContainedIn<OptionType,
+                                         ConfigOption::AvailableTypes>
+  void createConfigOption(const VectorOfKeysForJson& pathToOption,
+                          std::string_view optionDescription,
+                          OptionType* variableToPutValueOfTheOptionIn,
+                          std::optional<OptionType> defaultValue =
+                              std::optional<OptionType>(std::nullopt)) {
+    /*
+    In this case, we have an invalid path and can't create a `ConfigOption`.
+    Because we don't want to copy the corresponding exception out of
+    `verifyPathToConfigOption`, that would be to much code duplication, we
+    instead call it in such a way, that it will throw the corresponding
+    exception instead.
+    If we don't have this case, then the actual path check will be done by
+    `addConfigOption`. No reason, to do it double.
+    */
+    if (pathToOption.empty() ||
+        !std::holds_alternative<std::string>(pathToOption.back())) {
+      verifyPathToConfigOption(pathToOption, "");
+    }
+
+    if (defaultValue.has_value()) {
+      addConfigOption(
+          pathToOption,
+          ConfigOption(std::get<std::string>(pathToOption.back()),
+                       optionDescription, variableToPutValueOfTheOptionIn,
+                       defaultValue.value()));
+    } else {
+      addConfigOption(
+          pathToOption,
+          ConfigOption(std::get<std::string>(pathToOption.back()),
+                       optionDescription, variableToPutValueOfTheOptionIn));
+    }
```

Comment:
Funny thing: That is, how the constructor used to look like. But, apparently, the compiler can't indirectly convert a `T` to a `std::optional<T>` and I didn't want to burden future programmers with always typing `std::optional{}`, when they wanted to pass a default value, so I switched to having two constructors.

I just thought, that QOL-improvement were slightly more important than just having one constructor. What's your opinion?

Reply:
The only problem that I can imagine here is some implicit conversions etc. that make problems. But since
1. Those are somewhat evil anyway
2. Typically the ConfigOptions are created via such a Manager, so this is the most prominent position of this constructor usage.
So I am definitely for using only one constructor and then simplyfing this position.

Reply:
Well, you have more experience, so I'm going to change it back, like you wanted. See [8697880](https://github.com/ad-freiburg/qlever/pull/977/commits/86978800c0ec0c291f14036e81f672ee72b8c915).